openapi.go

```diff
@@ -251,14 +255,25 @@ func (route *RouteWithParams[Params, ResponseBody, RequestBody]) RegisterParams(
 	if typeOfParams.Kind() == reflect.Struct {
 		for i := range typeOfParams.NumField() {
 			field := typeOfParams.Field(i)
+			description, _ := field.Tag.Lookup("description")
 			if headerKey, ok := field.Tag.Lookup("header"); ok {
-				OptionHeader(headerKey, "string")(&route.BaseRoute)
+				OptionHeader(headerKey, description)(&route.BaseRoute)
 			}
 			if queryKey, ok := field.Tag.Lookup("query"); ok {
-				OptionQuery(queryKey, "string")(&route.BaseRoute)
+				switch field.Type.Kind() {
+				case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
+					reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64,
+					reflect.Float32, reflect.Float64:
+					OptionQueryInt(queryKey, description)(&route.BaseRoute)
+
+				case reflect.Bool:
```

Comment:
Ah I see this backpacks off the untyped params stuff too build the OpenAPI stuffs. 

One thing I hit while implementing the Array stuff locally. Our GoType field is well flat. There's a case where a user could provide an array of numbs `?mynum=1&mynum2` or string `?mystring=1&mystring2` and i guess even a bool (i don't know what someone could be doing with bools but 🤷). My initial thought is that we just switch `GoType` to 

```golang
type struct GoType {
  // integer, string, bool
  type string
  list bool
}
```

That was just my initial idea

Reply:
mmh i see